yarn-project/simulator/src/avm/opcodes/multi_scalar_mul.ts

```diff
@@ -64,11 +66,11 @@ export class MultiScalarMul extends Instruction {
     const scalarReadLength = numPoints * 2;
     // Consume gas prior to performing work
     const memoryOperations = {
-      reads: 1 + pointsReadLength + scalarReadLength /* points and scalars */,
+      reads: 1,
       writes: 3 /* output triplet */,
       indirect: this.indirect,
     };
-    context.machineState.consumeGas(this.gasCost(memoryOperations));
+    context.machineState.consumeGas(this.gasCost({ ...memoryOperations, dynMultiplier: pointsReadLength }));
```

Comment:
Simplified dynamic gas cost for MSM.... Shouldn't be a long term instruction anyway so no need to be super accurate, and no need to do something special that no other instruction does (deriving scaling factor rather than directly using the value read from memory).

Reply:
Didn't realize that the expected # `memoryOperations` are not actually impacting gas cost and are just checked vs actual